src/GlobalState.js

```diff
@@ -52,11 +101,107 @@ class Provider extends Component {
     return this.state.auth.loggedIn
   }
 
+  setUpWallet = async ({ action }) => {
+    let web3
+    // dappid is mandatory so will have throw away id for local usage.
+
+    let {
+      onboard,
+      networkState: { expectedNetworkId }
+    } = this.state
+    if (!onboard) {
+      let testid = 'c212885d-e81d-416f-ac37-06d9ad2cf5af'
+      onboard = Onboard({
+        dappId: BLOCKNATIVE_DAPPID || testid,
+        networkId: parseInt(expectedNetworkId),
+        walletCheck: walletChecks,
+        walletSelect: {
+          heading: 'Select a wallet to connect to Kickback',
+          description:
+            'To use Kickback you need an Ethereum wallet. Please select one from below:',
+          wallets: wallets
+        }
+      })
+      this.setState({ onboard })
+    }
+
+    let result = {
+      action,
+      error: false,
+      hasBalance: false,
+      expectedNetworkId
+    }
+
+    try {
+      // If user has chosen a wallet before then just use that.
+      const lastUsedWallet = LocalStorage.getItem(WALLET)
+      console.log('onboarding', lastUsedWallet)
+      const selected = await onboard.walletSelect(lastUsedWallet)
+
+      if (selected) {
+        const ready = await onboard.walletCheck()
+
+        if (ready) {
+          const walletState = onboard.getState()
+          let {
+            address,
+            network,
+            balance,
+            wallet,
+            mobileDevice
+            // appNetworkId
+          } = walletState
+
+          // Save this wallet provider for next login
+          LocalStorage.setItem(WALLET, wallet.name)
+
+          const web3 = new Web3(wallet.provider)
+          this.setState({ wallet, web3 })
+
+          result = {
+            mobileDevice,
+            network,
+            address,
+            ...result
+          }
+
+          result.correctNetwork = network === parseInt(expectedNetworkId)
+
+          // We want to know whether the users have any balances in their walllet
+          // but don't want to know how much they do.
+          result.hasBalance = parseInt(balance || 0) > 0
+          pollForBlocks(web3)
+        } else {
+          // Connection to wallet failed
+          LocalStorage.setItem(WALLET)
```

Comment:
Doesn't setItem need two arguments? What are you trying to set this to?

Reply:
I'm clearing LocalStorage here, I've changed it to removeItem to be more explicit